platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/EventVisitedMark.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2021-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.consensus;
+
+import com.swirlds.platform.internal.EventImpl;
+
+/**
+ * When searching for events in a graph, this instance is used to mark the ones that have already
+ * been visited in order to avoid visiting them more than once. It works by using a different number
+ * for each search.
+ */
+public class EventVisitedMark {
+    /**
+     * an event with this number is "marked", all others are "unmarked". Used by the
+     * ValidAncestorsIterator
+     */
+    private int currMark = ConsensusConstants.FIRST_MARK;
+
+    /** Called when a new search is starting to use a new mark value */
+    public void nextMark() {
+        // use only odd numbers so that 0 is always skipped
+        // 0 is the default mark value of an event, so we cannot use it
+        currMark += 2;
```

Comment:
explained above. we don't want to use the mark `0` to not confuse a new instance as a marked event in case of an overflow. makes sense?

Reply:
it seems a little awkward to have the special value of 0 defined in a different class, but influence the logic here. Perhaps the unmarked constant is better placed in this class rather than in `ConsensusConstants`.

Reply:
moved